typed/validate.go

```diff
@@ -185,35 +156,25 @@ func (v *validatingObjectWalker) doList(t *schema.List) (errs ValidationErrors)
 	return errs
 }
 
-func (v *validatingObjectWalker) visitMapItems(t *schema.Map, m *value.Map) (errs ValidationErrors) {
-	for i := range m.Items {
-		item := &m.Items[i]
-		pe := fieldpath.PathElement{FieldName: &item.Name}
-
-		if sf, ok := t.FindField(item.Name); ok {
-			v2 := v.prepareDescent(pe, sf.Type)
-			v2.value = item.Value
-			errs = append(errs, v2.validate()...)
-			v.finishDescent(v2)
-		} else {
-			v2 := v.prepareDescent(pe, t.ElementType)
-			v2.value = item.Value
-			errs = append(errs, v2.validate()...)
-			v2.doNode()
-			v.finishDescent(v2)
+func (v *validatingObjectWalker) visitMapItems(t *schema.Map, m value.Map) (errs ValidationErrors) {
+	m.Iterate(func(key string, val value.Value) bool {
```

Comment:
We have a lot of allocations because of the lambda escaping:
> `typed/validate.go:160:12: func literal escapes to heap`
Maybe about 36% of 'ConvertUnstructured/Endpoints/From`

Reply:
ah I missed this one. I think you can do this without making a closure:

```
func (v *validatingObjectWalker) visitMapItems(t *schema.Map, m value.Map) (errs ValidationErrors) {
  v.mapiterParent = t
  v.mapiterParent = m
  v.mapiterErrs = v.mapiterErrs[:0]
  m.Iterate(v.mapiter)
  return v.mapiterErrs
}

func (v *validatingObjectWalker) mapiter(key string, val value.Value) {
  // body of iteration function here
}
```

Reply:
Didn't work :-/

Reply:
Really? That's very sad

Reply:
How about this disgusting thing, does it work? (followup idea)

```
v.funcPtrStorage = func(key string, val value.Value) bool { ... }
m.Iterate(v.funcPtrStorage)
```